### DISCUSSION TITLE
疑似安全問題，郵件服務器自動使用未被允許的用戶名向位置的郵件地址發送郵件

### DISCUSSION THREAD

**Discussion by HoneyWhiteCloud** [Jinnrry/PMail#228] [General]
![image](https://github.com/user-attachments/assets/ecd452fe-460e-4420-8cec-dc8d1b188de6)
![image](https://github.com/user-attachments/assets/e485abca-4364-4d9d-a97d-30c682301b24)

（主題和發件人打碼部分為個人域名，郵件内容打碼處為片段數據，處於安全考慮將其打碼處理）

如圖所示，郵件服務器使用了隨機生成的用戶名（我的pmail服務端只有一個名為cloud的admin賬戶）稱向未知身份的郵件地址自動發送了含有疑似加密后的數據

服務器系統版本	Windows Server 2022 Datacenter
版本号	21H2
安装日期	‎2024/‎8/‎9
操作系统内部版本	20348.2849

安裝pmail時按照步驟開放了25, 110, 465, 995, 80 端口，並使用自定義有效的SSL證書
尚不明確是否為Pmail服務端存在安全問題，但本機在全盤掃描后并未發現任何異常程序

**Comment by HoneyWhiteCloud** [comment]
并且很奇怪，这封邮件出现在我的收件箱里面，而非发件箱。
![image](https://github.com/user-attachments/assets/d72396c0-5ac4-446b-9dd9-841a3217a096)

**Comment by Jinnrry** [comment]
1、这是别人发送给你的，不是你发给别人的
2、目前互联网的电子邮件（smtp）协议，是没有身份这个概念的，因此任何人都能伪造发件人。现目前主流的做法是spf和dkim认证，pmail也支持这两项认证，如果认证不过会在邮件前面加红色的感叹号

**Reply by HoneyWhiteCloud** [reply]
如果是別人發給我的，那爲何收件人為valxxx@megamail.com.br而非我的收件地址（pkyjzrnpc@domain.name）
是否為pmail存在bug會導致收件人和發件人混淆？
如未混淆，則認證不過也未在郵件前面加紅色感嘆號

**Reply by Jinnrry** [reply]
1、smtp没有身份的概念，发件人是可以随便修改随便设置的
2、你只有一个账号，因此所有收件都会投递给你这个账号，不管收件人是啥
3、收件人是对方指定的
4、没有原始日志信息，没法定位是否是漏洞

https://anchorety.github.io/2020/10/18/%E9%82%AE%E4%BB%B6%E5%AE%89%E5%85%A8%E4%B9%8B%E5%8F%91%E4%BB%B6%E4%BA%BA%E4%BC%AA%E9%80%A0/

邮件伪造是非常常见的诈骗方式，你这个的话，应该是批量发送利用某些服务器漏洞用的邮件，就和天天用弱密码去连接别人22 3306端口的操作一样

**Reply by HoneyWhiteCloud** [reply]
感谢你的回复

很抱歉未能提供相关日志，Pmail好像默认并不会储存日志（我在正常的设置流程走下来后，config.json内的日志记录等级为空，也未能找到相关日志记录的路径）

既然如此，pmail遇到此状况是否能说明我的服务器或存在安全漏洞或者有安全隐患？（事后依照一些教程尝试重新配置dmarc记录，是否有效还有待考察）

**Reply by HoneyWhiteCloud** [reply]
![image](https://github.com/user-attachments/assets/e2e23f1f-8c9d-4f77-b8af-b170e778494d)
在撰写回复和修改dmarc记录时，我又收到了来自未知发送者的邮件信息，但这次发件人不为我

**Reply by Jinnrry** [reply]
日志在哪里取决于你用什么方式部署项目，比如你用docker部署，就去docker里面找日志，PMail没有自己管理日志，而是交由服务管理进程管理日志的。

你的邮箱是公开过的吧，公开以后有这种扫描太正常了。

配置文件里面，把spamFilterLevel设置成1或者2可以拒收部分

https://github.com/Jinnrry/PMail/tree/master/server/hooks/spam_block

启用垃圾过滤插件，也能过滤部分。这2种反垃圾方式都启用基本上能解决

**Reply by HoneyWhiteCloud** [reply]
感谢您提供的帮助，我再多去钻研些
